.github/workflows/docker.yml

```diff
@@ -26,11 +27,35 @@ jobs:
         if: "${{ env.MY_TOKEN != '' }}"
         run: echo "token_defined=true" >> "$GITHUB_OUTPUT"
 
-  build:
-    name: Build
+  build-images:
+    name: Build Docker Images
     runs-on: ubuntu-latest
     needs: [check-token]
     if: needs.check-token.outputs.token_defined == 'true'
+    strategy:
+      matrix:
+        variant:
+          - name: complete
+            include_mongodb: true
+            include_postgresql: true
+            include_mariadb: true
+            tag_suffix: ""
+          - name: mongodb
+            include_mongodb: true
+            include_postgresql: false
+            include_mariadb: false
+            tag_suffix: "-mongodb"
+          - name: postgresql
+            include_mongodb: false
+            include_postgresql: true
+            include_mariadb: false
+            tag_suffix: "-postgresql"
+          - name: mariadb
```

Comment:
Maybe we should use mysql instead of mariadb?
After all, mariadb is mysql, just a different name
Most people still refer to mysql